app/src/main/java/com/google/android/sambadocumentsprovider/auth/AuthActivity.java

```diff
@@ -0,0 +1,169 @@
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.google.android.sambadocumentsprovider.R;
+import com.google.android.sambadocumentsprovider.SambaProviderApplication;
+import com.google.android.sambadocumentsprovider.ShareManager;
+import com.google.android.sambadocumentsprovider.base.AuthFailedException;
+import com.google.android.sambadocumentsprovider.base.OnTaskFinishedCallback;
+import com.google.android.sambadocumentsprovider.nativefacade.SmbClient;
+
+public class AuthActivity extends AppCompatActivity {
+  private static final String TAG = "AuthActivity";
+
+  private static final String SHARE_URI_KEY = "shareUri";
+
+  private EditText mSharePathEditText;
+  private EditText mDomainEditText;
+  private EditText mUsernameEditText;
+  private EditText mPasswordEditText;
+
+  private CheckBox mPinShareCheckbox;
+
+  private ProgressDialog progressDialog;
+
+  private ShareManager mShareManager;
+  private SmbClient mClient;
+
+  private final View.OnClickListener mLoginListener = new View.OnClickListener() {
+    @Override
+    public void onClick(View view) {
+      tryAuth();
+    }
+  };
+
+  private final OnTaskFinishedCallback<Void> callback = new OnTaskFinishedCallback<Void>() {
+    @Override
+    public void onTaskFinished(@Status int status, @Nullable Void item, @Nullable Exception e) {
+      progressDialog.dismiss();
+
+      if (status == SUCCEEDED) {
+        setResult(RESULT_OK);
+        finish();
+      } else {
+        Log.i(TAG, "Authentication failed: ", e);
+
+        final Context context = getApplicationContext();
+        if (e instanceof AuthFailedException) {
+          Toast.makeText(
+                  context,
+                  getApplicationContext().getString(R.string.credential_error),
+                  Toast.LENGTH_SHORT).show();
+        } else {
+          Toast.makeText(
+                  context,
+                  getApplicationContext().getString(R.string.credential_error),
+                  Toast.LENGTH_SHORT).show();
+        }
+      }
+    }
+  };
+
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    final Context context = getApplicationContext();
+    mShareManager = SambaProviderApplication.getServerManager(context);
+    mClient = SambaProviderApplication.getSambaClient(context);
+
+    Intent authIntent = getIntent();
+    String shareUri = authIntent.getStringExtra(SHARE_URI_KEY);
+
+    prepareUI(shareUri);
+  }
+
+  private void prepareUI(String shareUri) {
+    mSharePathEditText = (EditText) findViewById(R.id.share_path);
+    mUsernameEditText = (EditText) findViewById(R.id.username);
+    mDomainEditText = (EditText) findViewById(R.id.domain);
+    mPasswordEditText = (EditText) findViewById(R.id.password);
+
+    CheckBox mPasswodCheckbox = (CheckBox) findViewById(R.id.needs_password);
```

Comment:
Misspelled. Don't add `m` prefix to local variables.

Actually it kinda feels this checkbox is redundant. Maybe we can get rid of that entirely. Thoughts?